File: C#/Strategy.cs
using System;

//Strategy Interface
public interface IStrategy
{
    void Execute();
}

//Concrete Strategy A
public class ConcreteStrategyA : IStrategy
{
    public void Execute()
    {
        Console.WriteLine("Strategy A executed.");
    }
}

//Concrete Strategy B
public class ConcreteStrategyB : IStrategy
{
    public void Execute()
    {
        Console.WriteLine("Strategy B executed.");
    }
}

//Context Class
public class Strategy
{
    private IStrategy _strategy;

    public Strategy(IStrategy strategy)
    {
        _strategy = strategy;
    }

    public void SetStrategy(IStrategy strategy)
    {
        _strategy = strategy;
    }

    public void ExecuteStrategy()
    {
        _strategy.Execute();
    }
}


class Program
{
    static void Main()
    {
        //Create a Strategy object with a specific strategy
        Strategy strategyContext = new Strategy(new ConcreteStrategyA());
        strategyContext.ExecuteStrategy(); //Output: Strategy A executed.

        //Change the strategy at runtime
        strategyContext.SetStrategy(new ConcreteStrategyB());
        strategyContext.ExecuteStrategy(); //Output: Strategy B executed.
    }
}

File: Python/Collection/Strategy.py
from abc import ABC, abstractmethod

# Strategie Interface
class IStrategy(ABC):
    @abstractmethod
    def execute(self):
        pass

# Konkrete Strategie A
class ConcreteStrategyA(IStrategy):
    def execute(self):
        print("Strategie A ausgeführt.")

# Konkrete Strategie B
class ConcreteStrategyB(IStrategy):
    def execute(self):
        print("Strategie B ausgeführt.")

# Kontext-Klasse
class Strategy:
    def __init__(self, strategy: IStrategy):
        self._strategy = strategy

    def set_strategy(self, strategy: IStrategy):
        self._strategy = strategy

    def execute_strategy(self):
        self._strategy.execute()

# Beispiel der Benutzung
if __name__ == "__main__":
    # Strategie-Objekt erstellen
    strategy_context = Strategy(ConcreteStrategyA())
    strategy_context.execute_strategy()  # Ausgabe: Strategie A ausgeführt.

    # Strategie zur Laufzeit ändern
    strategy_context.set_strategy(ConcreteStrategyB())
    strategy_context.execute_strategy()  # Ausgabe: Strategie B ausgeführt.
